src/hostapi/asio/pa_asio.cpp

```diff
@@ -1304,6 +1304,16 @@ PaError PaAsio_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex
                 continue;
             }
 
+            // Blacklist dynamic loading of JackRouter in 32-bit applications.
+            // This causes an instant crash due to a long-standing address translation error.
+            //   https://github.com/jackaudio/jack2/issues/332
+
+            if (sizeof(void*)>4 && strcmp (names[i],"JackRouter") == 0)
```

Comment:
Won't this also block it in 64bit apps? Since a pointer is 8B there.

Reply:
My memory is vague but it may actually be the comment that's incorrect.  As I recall, the defect in JackRouter involves attempting to store a 64-bit pointer from the application in a 32-bit value, later expanding it into an invalid pointer.  Some (re-)testing may be in order.

Reply:
Confirmed based on discussion here:  https://github.com/jackaudio/jack2/issues/275

64-bit applications will crash immediately upon opening the JackRouter ASIO driver.  I'll amend the comment.

Reply:
I believe that [Jack 1.9.17 on Windows](https://github.com/jackaudio/jack2/releases/tag/v1.9.17) provides a new, recompiled version of JackRouter ASIO driver, which I believe doesn't have this issue anymore. I can't find the relevant discussion thread though...

Reply:
Unfortunately older installations of JACK are still extant on users' machines.  Considering this problem creates an immediate crash in Pa_Initialize where it does occur, the driver should remain blacklisted for a while (unless there's a way to detect its version number here, which might be out of scope).

Reply:
This is tangential to this PR, but I got PortAudio to build on Windows with the JACK host API: https://github.com/PortAudio/portaudio/pull/618 Considering the ASIO backend explicitly prevents using JackRouter, I think this would be the only way for PortAudio applications to use JACK on Windows.

It builds but it doesn't seem to work on Windows. I have not had a chance to debug it yet. Help would be appreciated if you are interested.